lib/smart_proxy_reports/spooled_http_client.rb

```diff
@@ -52,37 +52,44 @@ def stop
       @worker.join
     end
 
+    def connect(http, factory, type, endpoint)
```

Comment:
This method is used to send both reports and facts. Which is correct I guess, but needs some modifications if I'm right in the next comment.

 - If I'm right then current approach *might* mess the state of a host(s).
 - This method should be used once since it goes through everything in `todo` (and if I read correctly then we store there reports + facts) directory anyway.
 - The method doesn't need `type` nor `endpoint` as params if we do the above/below stuff.
 - This method should send a report with its corresponding facts in the iteration, e.g. `report-same-id` should be sent with `facts-same-id`.
 - Thus `processed` variable will contain the actual number of processed reports and its facts.

Reply:
> approach might mess the state of a host(s)

Can you be more specific?

I understand that if there are multiple reports queued with respective fact files, the current approach will send first all facts and then all reports. But what kind of inconsistency are we talking about? But I also have feeling that we should respect the order. More importantly, order of multiple reports - that could be a problem.

Reply:
Yeah, sorry for not being fully explicit. My concern is more about the order as you're saying.